### PR TITLE
[13.0][MIG] product_multi_company

### PR DESCRIPTION
**Pull request by JoanSForgeFlow**
**product_multi_company** migration to v13.
Depends on:
- #197 

cc ~ @LoisRForgeFlow @JordiBForgeFlow

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
@JoanSForgeFlow Can you check travis?

File: product_multi_company/models/product_category.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    def get_user_companies(self):
+        return self.env.user.company_ids.ids
+
+    total_route_ids = fields.Many2many(
+        domain=[
+            "|",
+            ("company_id", "=", False),
+            ("company_id", "in", "get_user_companies"),
+        ]
```
Comment (LoisRForgeFlow) [inline product_multi_company/models/product_category.py]: I think the proper way would be to create a new field computed field `allowed_route_ids` and then update this domain to be: `domain="[('id', 'in', allowed_route_ids)]

Comment (AaronHForgeFlow) [reply product_multi_company/models/product_category.py]: Tried that but got recursion error as long as allowed_route_ids needed route_ids in order to calculate those :( Now I think it is working, but still checking

**Comment by LoisRForgeFlow** [comment]
@AaronHForgeFlow Great! looks much better now! Could you attend travis and squash your commits?

**Comment by AaronHForgeFlow** [comment]
Basically, the default method adds a company to https://github.com/OCA/multi-company/blob/13.0/account_invoice_inter_company/tests/inter_company_invoice.xml#L383 so, putting that `ref=""` is useless when product_multi_company is installed.

Besides that, in case company_ids is not set. The company_id should be False.

I am not sure if creating separate PRs for that...

**Comment by AaronHForgeFlow** [comment]
@JoanSForgeFlow could you remove the "WIP" from the subject? :)

**Comment by AaronHForgeFlow** [comment]
Finally, the extra check in base_multi_company module was unnecessary, and I removed it. In case company_ids is not set in the model it goes to the "else" in the compute method and no company_id is set.

**Review by LoisRForgeFlow — APPROVED** [review]
Great, Thanks for the work!

**Review by AaronHForgeFlow — APPROVED** [review]
Functional test + code review :innocent:

**Pull request merged**